client/signup/steps/site-topic/index.jsx

```diff
@@ -133,37 +131,38 @@ class SiteTopicStep extends Component {
 					subHeaderText={ commonSubHeaderText }
 					fallbackSubHeaderText={ commonSubHeaderText }
 					signupProgress={ this.props.signupProgress }
-					stepContent={ this.renderContent( topicLabel, commonPlaceholder ) }
+					stepContent={ this.renderContent() }
 				/>
 			</div>
 		);
 	}
 }
 
 const mapDispatchToProps = ( dispatch, ownProps ) => ( {
-	submitSiteTopic: siteTopic => {
+	submitSiteTopic: ( { vertical_name, vertical_slug }, is_user_input_vertical ) => {
 		const { flowName, goToNextStep } = ownProps;
 
 		dispatch(
 			recordTracksEvent( 'calypso_signup_actions_submit_site_topic', {
-				value: siteTopic,
+				value: vertical_slug,
+				is_user_input_vertical,
```

Comment:
I think we need to store this in the state tree since coming to this step by hitting the Back button from a later step always sets `is_user_input_vertical ` to `true`, the default value.